server/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -342,13 +348,28 @@ public MatchQueryBuilder zeroTermsQuery(MatchQuery.ZeroTermsQuery zeroTermsQuery
         return this;
     }
 
+    /**
+     * Sets query to use when single term synonyms overlap positions
+     * {@link MatchQuery.SynonymQueryStyle#BLENDED_TERMS}, but can be set to
+     * {@link MatchQuery.SynonymQueryStyle#MOST_TERMS} or {@link MatchQuery.SynonymQueryStyle#BEST_TERMS} instead.
+     */
+    public MatchQueryBuilder synonymQueryStyle(MatchQuery.SynonymQueryStyle synQueryStyle) {
+        if (synQueryStyle == null) {
+            throw new IllegalArgumentException("[" + NAME + "] requires synQueryStyle to be non-null");
```

Comment:
wouldn't it be better if the exception uses `synonym_query_style` instead of `synQueryStyle`. It would be more clear, I think.